src/library/compiler/ir_interpreter.cpp

```diff
@@ -1035,7 +1035,18 @@ class interpreter {
                 dec_ref(r);
                 symbol_cache_entry e = lookup_symbol(decl);
                 if (e.m_addr) {
-                    *((object **)e.m_addr) = o;
+                    switch (decl_type(e.m_decl)) {
+                        case type::Float: *((double*)e.m_addr) = unbox_float(o); break;
+                        case type::UInt8: *((uint8*)e.m_addr) = unbox(o); break;
+                        case type::UInt16: *((uint16*)e.m_addr) = unbox(o); break;
+                        case type::UInt32: *((uint32*)e.m_addr) = unbox_uint32(o); break;
+                        case type::UInt64: *((uint64*)e.m_addr) = unbox_uint64(o); break;
+                        case type::USize: *((size_t*)e.m_addr) = unbox_size_t(o); break;
+                        case type::Object:
+                        case type::TObject:
+                        case type::Irrelevant: *((object **)e.m_addr) = o; break;
+                        default: lean_unreachable();
+                    }
```

Comment:
I think this code is correct, or at least as correct as it was before, but AFAICT this entire branch is unreachable: you can replace this entire block with `throw;` and it still passes all the tests. It is handling the case where you call an initializer on a declaration that has already been initialized, which AFAIK is not legal.